miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditServiceRecordController.java

```diff
@@ -0,0 +1,147 @@
+  
+  @Autowired
+  private RequestManager requestManager;
+
+  @Autowired
+  private FilesManager filesManager;
+
+  @Autowired
+  private DataObjectFactory dataObjectFactory;
+
+  public void setRequestManager(RequestManager requestManager) {
+    this.requestManager = requestManager;
+  }
+
+  public void setFilesManager(FilesManager filesManager) {
+    this.filesManager = filesManager;
+  }
+
+  public void setDataObjectFactory(DataObjectFactory dataObjectFactory) {
+    this.dataObjectFactory = dataObjectFactory;
+  }
+  
+  public Map<Integer, String> populateServiceRecordFiles(SequencerServiceRecord record) throws IOException {
+    if (record.getId() != AbstractSequencerServiceRecord.UNSAVED_ID) {
+      Map<Integer, String> fileMap = new HashMap<Integer, String>();
+      for (String s : filesManager.getFileNames(SequencerServiceRecord.class, String.valueOf(record.getId()))) {
+        fileMap.put(s.hashCode(), s);
+      }
+      return fileMap;
+    }
+    return Collections.emptyMap();
+  }
+  
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> maxLengths() throws IOException {
+    return requestManager.getServiceRecordColumnSizes();
+  }
+
+  @RequestMapping(value = "/{recordId}", method = RequestMethod.GET)
+  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, ModelMap model) throws IOException {
+    SequencerServiceRecord sr = requestManager.getSequencerServiceRecordById(recordId);
+    if (sr != null) {
+      model.put("serviceRecord", sr);
+      model.put("serviceRecordFiles", populateServiceRecordFiles(sr));
+    } else {
+      throw new IOException("Cannot retrieve the requested Service Record");
+    }
+    return new ModelAndView("/pages/editServiceRecord.jsp", model);
+  }
+  
+  @RequestMapping(value = "/new/{sequencerReferenceId}", method = RequestMethod.GET)
+  public ModelAndView newServiceRecord(@PathVariable(value = "sequencerReferenceId") Long sequencerReferenceId, ModelMap model) throws IOException {
+    SequencerReference sequencer = requestManager.getSequencerReferenceById(sequencerReferenceId);
+    if (sequencer == null) {
+      throw new IOException("Cannot retrieve the requested Sequencer Reference");
+    }
+    SequencerServiceRecord record = dataObjectFactory.getSequencerServiceRecord();
+    record.setSequencerReference(sequencer);
+    model.put("serviceRecord", record);
+    return new ModelAndView("/pages/editServiceRecord.jsp", model);
+  }
+  
+  @RequestMapping(method = RequestMethod.POST)
+  public String processSubmit(@ModelAttribute("serviceRecord") SequencerServiceRecord record, ModelMap model, SessionStatus session)
+      throws IOException {
+    try {
+      requestManager.saveSequencerServiceRecord(record);
+      session.setComplete();
+      model.clear();
+      return "redirect:/miso/stats/sequencer/servicerecord/" + record.getId();
+    } catch (IOException ex) {
```

Comment:
I guess I am a little uncomfortable with the use of checked exceptions here.
This exception goes down through the request manager and (I think) tells us whether a user is admin OR `SequencerServiceRecord` that is being sent through is `null`.
Since we don't attempt to recover and exceptions get logged anyways we have really done nothing useful with this exception.
Here are some other ideas...
######Options include:
* Not throwing the exceptions in the first place and just doing the logging.
* Throwing *RuntimeException* instead so you don't have to bother catching and you get the logging for free.
* Attempting some sort of recovery if this exception is caught (forward to `404` / `400` page).

Personally I'm all about checking the inputs in the API before they ever make it to request manager.  I think they call it *sanitizing your inputs*.  That way you don't even try to save if things don't look right, you can just redirect to a friendly "Nope 404 / 400" page.  It's a better user experience I think.